dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/teis/RelationshipsTest.java

```diff
@@ -149,6 +148,37 @@ public void duplicateRelationshipsShouldNotBeAdded()
             .body( "importCount.ignored", equalTo( 1 ) );
     }
 
+    @Test
+    public void softDeletedRelationshipsShouldNotBeUpdated()
+    {
+        // create a relationship
+        JsonObject object = relationshipActions
+            .createRelationshipBody( "xLmPUYJX8Ks", "trackedEntityInstance", teis.get( 0 ), "trackedEntityInstance",
+                teis.get( 1 ) );
+
+        ApiResponse response = relationshipActions.post( object );
+
+        response.validate().statusCode( 200 );
+        createdRelationship = response.extractUid();
+
+        relationshipActions.softDelete( createdRelationship );
+
+        // Update soft deleted relationship
+        object.addProperty( "relationship", createdRelationship );
+        response = relationshipActions.post( object );
+
+        response.validate().statusCode( 409 )
+            .body( "status", equalTo( "ERROR" ) )
+            .body( "response.status", equalTo( "ERROR" ) )
+            .body( "response.ignored", equalTo( 1 ) )
+            .body( "response.total", equalTo( 1 ) )
+            .rootPath( "response.importSummaries[0]" )
+            .body( "status", equalTo( "ERROR" ) )
+            .body( "description", Matchers.containsString(
+                "Relationship '" + createdRelationship + "' is already deleted and cannot be modified." ) )
```

Comment:
Why did you go for an e2e test here? Could we not test this in a unit test of AbstractRelationshipService

I just imagine making a change to AbstractRelationshipService that breaks this e2e test and how long it takes me to realize that 😅 might be 20min vs < 1min

Reply:
I agree with you.
I think I am quite skeptical when I am working with the old importer, I thought a unit test was going to be much more difficult.